src/test/java/io/vertx/ext/asyncsql/tx/IsolationLevelTest.java

```diff
@@ -0,0 +1,387 @@
+//      select current_setting('transaction_isolation');
+//      COMMIT;
+//      SELECT current_setting('transaction_isolation');
+
+      final AtomicReference<TransactionIsolation> ref = new AtomicReference<>();
+
+      conn.getTransactionIsolation(ar1 -> {
+        ensureSuccess(context, ar1);
+        // base line
+        ref.set(ar1.result());
+        conn.setTransactionIsolation(TransactionIsolation.READ_UNCOMMITTED, ar2 -> {
+          ensureSuccess(context, ar2);
+          conn.setAutoCommit(false, ar3 -> {
+            ensureSuccess(context, ar3);
+            conn.getTransactionIsolation(ar4 -> {
+              ensureSuccess(context, ar4);
+              assertEquals(TransactionIsolation.READ_UNCOMMITTED, ar4.result());
+              conn.commit(ar5 -> {
+                ensureSuccess(context, ar5);
+                conn.setAutoCommit(true, ar6 -> {
+                  ensureSuccess(context, ar6);
+                  conn.getTransactionIsolation(ar7 -> {
+                    ensureSuccess(context, ar7);
+                    assertEquals(ref.get(), ar7.result());
+                    async.complete();
+                  });
+                });
+              });
+            });
+          });
+        });
+      });
+    });
+  }
+
+  private void setupTxTestTable(SQLConnection conn, Handler<AsyncResult<Void>> handler) {
+    conn.execute("BEGIN",
+      ar -> conn.execute("DROP TABLE IF EXISTS test_txtable",
+        ar2 -> conn.execute("CREATE TABLE test_txtable (id BIGINT, val BIGINT)",
+          ar3 -> conn.update("INSERT INTO test_txtable (id, val) VALUES (1, 8), (2, 8)",
+            ar4 -> conn.execute("COMMIT", handler)))));
+  }
+
+  interface IsolationHandler {
+    void handle(SQLConnection conn1, SQLConnection conn2);
+  }
+
+  private void runIsolationLevelTest(TestContext context, IsolationHandler handler) {
+    client.getConnection(ar -> {
+      if (ar.failed()) {
+        context.fail(ar.cause());
+        return;
+      }
+
+      // Create table
+      conn = ar.result();
+      setupTxTestTable(conn, ar2 -> {
+        if (ar2.failed()) {
+          context.fail(ar2.cause());
+          return;
+        }
+
+        // acquire 2 new connections
+        client.getConnection(ar3 -> {
+          if (ar3.failed()) {
+            context.fail(ar3.cause());
+            return;
+          }
+
+          // set auto commit off to connection
+          final SQLConnection conn1 = ar3.result();
+
+          client.getConnection(ar5 -> {
+            if (ar5.failed()) {
+              context.fail(ar5.cause());
+              return;
+            }
+
+            // set auto commit off to connection
+            final SQLConnection conn2 = ar5.result();
+
+            // close the setup connection
+            conn.close();
+
+            // run test
+            handler.handle(conn1, conn2);
+          });
+        });
+      });
+    });
+  }
+
+  @Test
+  @Ignore
```

Comment:
The reason is that PostgreSQL and MySQL handle this in different ways. The most obvious case is `read uncommitted` where PostgreSQL does not really read uncommitted data but does accept the isolation level. With MySQL it depends on the engine innodb does support it.

So these test must be rdmbs specific...